neuroml/nml/helper_methods.py

```diff
@@ -1065,7 +1093,10 @@ def get_by_id(self,id):
             return None
         all_ids = []
         for ms in self.member_data_items_:
-            mlist = self.__getattribute__(ms.name)
+            mlist = getattr(self, ms.get_name())
+            # TODO: debug why this is required
+            if mlist is None:
```

Comment:
@pgleeson not sure why this is required, but without this, it doesn't continue iterating over the next elements of `self.member_data_items_`, just stops at the first one :worried: